pkg/api/core/v1/models/env.go

```diff
@@ -1,5 +1,8 @@
 package models
 
+// __ATTENTION__ Functionally identical to AppSettings, AppSettingList
```

Comment:
Do we really need to warn people about this?

Reply:
There are ways to DRY this if we want to. E.g. we can create a struct that implements all these methods and add it as a "promoted field" both in `EnvVariable` and `AppSetting`

```
type EnvVariable struct {
  KeyValueObject
}

type AppSetting struct {
  KeyValueObject
}
```

though given the only methods/fields we need are those of the new struct, maybe we don't need the wrapping structs at all. 

Let's think about it outside this PR.

Reply:
> Do we really need to warn people about this?

People in general ? No.
The future maintainers, us or others ? IMHO yes.

Agreed to think in separate issue and PR.